addresses-scheme/readme.md

```diff
@@ -70,3 +70,192 @@ In [this cryptic comment](https://github.com/ipfs/in-web-browsers/issues/28#issu
+  if  [[ $1 == dweb:* ]] ;
+  then
+    prefix="dweb:"
+    internal_path=${1#$prefix}
+    file_getter="ipfs cat $internal_path"
+  else
+    file_getter="cat $1"
+  fi
+  echo $file_getter
+}
+
+file1=`get_multiaddr_or_normal_file_getter $1`
+file2=`get_multiaddr_or_normal_file_getter $2`
+
+diff <($file1) <($file2)
+````
+
+Another example that @timthelion came up with was the case of a hypothetical markdown to pdf utility which supports ipfs. Imagine we have a utilty named `md-to-pdf-with-ipfs-support` and we pass it a markdown file like so:
+
+**atic.md**
+````
+Stuff I found in my attic
+----------------------------------
+
+![An old box of rocks](/ipfs/QmdyWzsrBvSkPYPU1ScBpwzfCcegzbc6c2hkEBLLJ6VcPV)
+
+An old box of rocks.
+
+![A can of oil for water-proofing leather](/ipfs/QmUPC5xbVtu6NxMwFBtmWVjrVM3XffuPtSMLpmDFGfTaKG)
+````
+
+````
+$ md-to-pdf-with-ipfs-support attic.md > attic.pdf
+````
+
+How is this utility supposed to recognise which paths to resolve using `ipfs` and which paths to resolve normally? @flyingzumwalt suggests that prefixing `dweb:` for ipfs paths resolves this ambiguity and @timthelion concures.
+
+````
+Stuff I found in my attic
+----------------------------------
+
+![An old box of rocks](dweb:/ipfs/QmdyWzsrBvSkPYPU1ScBpwzfCcegzbc6c2hkEBLLJ6VcPV)
+
+An old box of rocks.
+
+![A can of oil for water-proofing leather](dweb:/ipfs/QmUPC5xbVtu6NxMwFBtmWVjrVM3XffuPtSMLpmDFGfTaKG)
+````
+
+@timthelion does not see resolving `/ipfs/` paths to be difficult merely for computers. As a human he is confused as well. When he sees `/ipfs` written on his local machine, he imagines that this means that the ipfs filesystem must be mounted in root for that path to be accessable. After reading the [multiaddr specs](https://github.com/multiformats/multiaddr) he had the feeling that this system would create unimaginable clutter on his system.
+
+"""
+Today, when I do `ls /` I get:
+
+````
+$ ls /
+bin/   dev/  home/        lib/    media/  opt/   root/  sbin/  sys/  usr/  vmlinuz@
+boot/  etc/  initrd.img@  lib64/  mnt/    proc/  run/   srv/   tmp/  var/
+````
+
+With the paths proposed by multiaddr I would instead see:
+
+````
+$ ls /
+bin/ bitcoin/ boot/ dev/ dns/ dns4/ dns6/ etc/ home/ http/ https/ initrd.img@ ipfs/ lib/ lib64/
+ libp2p-circuit-relay/ libp2p-webrtc-direct/ libp2p-webrtc-star/ media/ mnt/ onion/ opt/ p2p/ 
+proc/ root/ run/ sbin/ sctp/ srv/ sys/ tmp/ udt/ unix/ usr/ utp/ var/ vmlinuz@
+````
+[multiaddr-protocols](https://github.com/multiformats/multiaddr/blob/master/protocols.csv)
+"""
+
+While this may be a misunderstanding on his part, other users may be confused as well.
+
+@timthelion suggests, that if the ipfs devs insist on mixing special paths into the POSIX `/` namespace, they at least use a subdirectory and not polute all of the filesystem root `/`.
+
+### @timthelion's compromise: place all `multiaddr` filesystems under a special directory `/webfs/` and treat the prefix `/webfs/` as a magic path prefix
```

Comment:
I kinda like this aesthetically. No matter what prefix is picked, it looks better than anything with `://` 
`/webfs/ipfs/QmT272yei1Zn1eAUq5P9nZyeaKP4oJmVv7CbYvUPyk3aLj/hobby.jpg`
`/dweb/ipfs/QmT272yei1Zn1eAUq5P9nZyeaKP4oJmVv7CbYvUPyk3aLj/hobby.jpg`
`/x/ipfs/QmT272yei1Zn1eAUq5P9nZyeaKP4oJmVv7CbYvUPyk3aLj/hobby.jpg`